File: vue-vben-admin/apps/web-antd/src/api/growth/habit.ts
import type { BaseEntity, HabitStatus, PageQuery, PageResult } from './types';

import { requestClient } from '#/api/request';

export interface Habit extends BaseEntity {
  userId?: string;
  name: string;
  habitType: string;
  description?: null | string;
  targetFrequency: string;
  status: HabitStatus;
  currentStreak: number;
  longestStreak: number;
  totalCheckIns: number;
  lastCheckInDate?: null | string;
}

export interface HabitDetail extends Habit {
  recentCheckIns: Array<{
    id: string;
    checkInDate: string;
    remark?: null | string;
  }>;
}

export interface HabitQuery extends PageQuery {
  habitType?: string;
  keyword?: string;
  status?: HabitStatus;
}

export interface SaveHabitInput {
  description?: null | string;
  habitType: string;
  name: string;
  status?: HabitStatus;
  targetFrequency?: string;
}

export async function getHabitPageApi(params: HabitQuery) {
  return requestClient.get<PageResult<Habit>>('/growth/habits', { params });
}

export async function getHabitApi(id: string) {
  return requestClient.get<HabitDetail>(`/growth/habits/${id}`);
}

export async function createHabitApi(data: SaveHabitInput) {
  return requestClient.post<Habit>('/growth/habits', data);
}

export async function updateHabitApi(id: string, data: SaveHabitInput) {
  return requestClient.put<Habit>(`/growth/habits/${id}`, data);
}

export async function checkInHabitApi(id: string, remark?: string) {
  return requestClient.post<Habit>(`/growth/habits/${id}/check-in`, { remark });
}

export async function updateHabitStatusApi(id: string, status: HabitStatus) {
  return requestClient.put<Habit>(`/growth/habits/${id}/status`, { status });
}

export async function deleteHabitApi(id: string) {
  return requestClient.delete(`/growth/habits/${id}`);
}

File: WebApplication1/WebApplication1/Features/Growth/Entities/Habit.cs
using WebApplication1.Shared;

namespace WebApplication1.Features.Growth.Entities;

public class Habit : EntityBase
{
    public Guid? UserId { get; set; }

    public string Name { get; set; } = string.Empty;

    public string HabitType { get; set; } = string.Empty;

    public string? Description { get; set; }

    public string TargetFrequency { get; set; } = "每天";

    public int Status { get; set; } = 1;

    public int CurrentStreak { get; set; }

    public int LongestStreak { get; set; }

    public int TotalCheckIns { get; set; }

    public DateOnly? LastCheckInDate { get; set; }

    public ICollection<HabitCheckIn> CheckIns { get; set; } = new List<HabitCheckIn>();
}

public class HabitCheckIn : EntityBase
{
    public Guid HabitId { get; set; }

    public DateOnly CheckInDate { get; set; }

    public string? Remark { get; set; }

    public Habit Habit { get; set; } = null!;
}